setup.py

```diff
@@ -13,22 +13,41 @@
 # limitations under the License.
 
 import pathlib
+import functools
+import operator
+
 from setuptools import find_packages, setup
 
 __version__ = ''
 exec(open('recirq/_version.py').read())
 assert __version__, 'Version string cannot be empty'
 
-required_packages = pathlib.Path('requirements.txt').read_text().split('\n')
-INSTALL_PACKAGES = [pkg for pkg in required_packages if pkg and not pkg.startswith('#')]
+
+def _parse_requirements(path: pathlib.Path):
+    lines = [line.strip() for line in path.read_text().splitlines() if line]
+    return [line for line in lines if not line.startswith('#')]
+
+
+install_requires = _parse_requirements(pathlib.Path('requirements.txt'))
+extras_require = [
+    'otoc', 'qaoa', 'optimize', 'hfvqe', 'fermi_hubbard'
```

Comment:
Is there a good way to auto-detect this (or alternatively, for keeping this up to date if we add more experiments)?

Reply:
Yes, I first wrote it to glob/regex for extra-requirements.txt and I could set it to that. However: for my next thing I want to add CI jobs that tests each subpackage in isolation. Autogenerating CI jobs would likely be more complicated than its worth. Especially since there will be an isolated "vanilla deps" build that will run all tests _not_ in the named extra-requires subpackages which would fail if they did indeed have extra requirements.

Reply:
Ok, SGTM, I already approved this PR.  Feel free to merge when ready.